pkg/drivers/vfkit/vfkit.go

```diff
@@ -193,6 +214,42 @@ func (d *Driver) Create() error {
 }
 
 func (d *Driver) Start() error {
+	var helperSock, vfkitSock *os.File
+	var err error
+
+	if d.VmnetHelper != nil {
+		helperSock, vfkitSock, err = vmnet.Socketpair()
+		if err != nil {
+			return err
+		}
+		defer helperSock.Close()
+		defer vfkitSock.Close()
+
+		if err := d.VmnetHelper.Start(helperSock); err != nil {
+			return err
+		}
+
+		d.MACAddress = d.VmnetHelper.GetMACAddress()
+	}
+
+	if err := d.startVfkit(vfkitSock); err != nil {
+		d.stopVmnetHelper()
+		return err
+	}
+
+	if err := d.setupIP(d.MACAddress); err != nil {
+		d.stopVmnetHelper()
+		return err
+	}
+
+	log.Infof("Waiting for VM to start (ssh -p %d docker@%s)...", d.SSHPort, d.IPAddress)
+
+	return WaitForTCPWithDelay(fmt.Sprintf("%s:%d", d.IPAddress, d.SSHPort), time.Second)
+}
+
+// startVfkit starts the vfkit child process. If vfkitSock is non nil, vfkit is
+// connected to the vmnet network via the socket instead of "nat" network.
+func (d *Driver) startVfkit(vfkitSock *os.File) error {
```

Comment:
might be clearer to rename `vfkitSock` to `vmnetSock` in this method.

Reply:
The name come from Start() when we want to make it clear that one socket is for the helper, and one for vfkit. Keeping the same name in startVfkit() to make it easier to follow the code. It is very confusing when using different names to the same thing in different fuctions.

Maybe we can name both vmnetHelperSock and vmnetVfkitSock?

Reply:
This is minor, fine with me if you want to keep the current naming. `vmnetHelperSock`/`vmnetVfkitSock` is good too even if a bit on the "long" side.

Reply:
Keeping the current names then.